docs/source/integrations/plugins.mdx

```diff
@@ -3,40 +3,52 @@ title: Creating Apollo Server plugins
 description: Extend Apollo Server with custom functionality
 ---
 
-> **New in Apollo Server 3:** All plugin lifecycle methods are `async`, _except for [`willResolveField`](./plugins-event-reference/#willresolvefield) and [`schemaDidLoadOrUpdate`](./plugins-event-reference/#schemadidloadorupdate)_.
+> All plugin lifecycle methods are `async`, _except for [`willResolveField`](./plugins-event-reference/#willresolvefield) and [`schemaDidLoadOrUpdate`](./plugins-event-reference/#schemadidloadorupdate)_.
 
 You can create your own Apollo Server plugins to perform custom operations in response to certain events. For example, a basic logging plugin might log the GraphQL query string associated with each request that's sent to Apollo Server.
 
 ## The anatomy of a plugin
 
+> If you're using TypeScript to create a plugin, your plugins should implement the [`ApolloServerPlugin` interface](https://github.com/apollographql/apollo-server/blob/cc447c5687f67938efb976d7aa282f7a95e313db/packages/server/src/externalTypes/plugins.ts#L34).
```

Comment:
I think the docs should mention somewhere that if your plugin ever wants to read `requestContext.contextValue`, then you need to declare your plugin as `ApolloServerPlugin<YourContextType>`.

Maybe this could be part of something on the event reference page that actually documents GraphQLRequestContext? We don't document that or GraphQLServerContext.